src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1108,7 +1108,7 @@ private function registerValidationConfiguration(array $config, ContainerBuilder
 
         $validatorBuilder = $container->getDefinition('validator.builder');
 
-        if (class_exists(LegacyTranslatorProxy::class)) {
+        if (class_exists(Translator::class) && class_exists(LegacyTranslatorProxy::class)) {
```

Comment:
we use this check everywhere else in the extension to decide whether the Translation component is installed

Reply:
I saw (class_exists(LegacyTranslatorProxy::class)) in multiple places, should be all of them fixed?

Reply:
I didn't do that to avoid merge conflicts (3.4 is still maintained but affected by this bug).

Reply:
You mean *not* affected right? Because we remove the translator dep un 4.2. Makes sense.

Reply:
Comment correct yes, and while doing that I realised that I completely misread @goetas' comment.

@goetas Yes, there are other usages of this class, but I do not see that any of them can happen when the Translation component is not installed.

Reply:
wait, this should check for TranslatorInterface instead, from the component!